File: modules/developer_manual/examples/core/scripts/go/decline-pending-share.go
package main

import (
	"fmt"
	"io/ioutil"
	"log"
	"net/http"
	"os"
)

func main() {
	serverUri := "{oc-examples-server-url}/ocs/v1.php/apps/files_sharing/api/v1"
	username := "admin"
	passwd := "admin"

	client := &http.Client{}

	req, err := http.NewRequest("DELETE", fmt.Sprintf("%s/%s", serverUri, "shares/pending/<share_id>"), nil)
	if err != nil {
		log.Print(err)
		os.Exit(1)
	}

	req.SetBasicAuth(username, passwd)

	resp, err := client.Do(req)
	if err != nil {
		log.Fatal(err)
	}

	bodyText, err := ioutil.ReadAll(resp.Body)
	fmt.Println(string(bodyText))
}

File: modules/developer_manual/examples/core/scripts/ruby/decline-pending-share.rb
require 'net/http'
require 'uri'

base_uri = '{oc-examples-server-url}/ocs/v1.php/apps/files_sharing/api/v1/'
uri = URI(base_uri + "shares/pending/<share_id>")

Net::HTTP.start(uri.host, uri.port, :use_ssl => uri.scheme == 'https') do |http|
  req = Net::HTTP::Delete.new uri
  req.basic_auth 'your.username', 'your.password'
  res = http.request req

  puts res.body
end

